rfcs/0132-use-meson-to-build-nix.md

```diff
@@ -0,0 +1,121 @@
+---
+feature: use_meson_to_build_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for the reference implementation of
+Nix.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the reference implementation of Nix evaluator and its companion
+toolset generated from the Nix source tree are built using an autotools-like
+script.
+
+This quasi-autotools script became clunky and plastered, and consequently hard
+to understand, modify, improve and port to other systems besides Linux.
```

Comment:
Do we have information on what systems we expect to be able to support with Meson that are currently unsupported? Ideally we will turn this into a theoretical benefit into a concrete one, especially since autotools is infinitely portable (just incredibly painful to do so).

Reply:
TBH I think the GNU Make `lib` code is quite nice, but the autoconf is not so nice, and GNU make isn't as good a standard as ninja.

So I would emphasize:

 - Autoconf is bad
 - Ninja runs on more platforms, is the preferred build system lingua franca

Reply:
@kevincox the idea is to expand support for *BSD *XNU minix and maybe even haiku or NT. the end goal is to get nix to a point where it can be bootstrapped on any platform, not just gnu/linux

Reply:
@Ericson2314 i would agree. its possible to create good looking makefiles,ad but autotools isn't good for readability, maintainability, cross platform support, and also greatly increases the amount of dependencies needed in the toolchain.

i think in the long term once meson is stable, it would be a good idea to replace the autotools with makefiles (can be used on most platforms) or cmake (more powerful but slightly harder to bootstrap).

Reply:
FWIW the "We want Nix to work reliably everywhere" heading from  https://www.tweag.io/blog/2022-08-04-tweag-and-nix-future/ to me indicates Windows ought to be important.

Reply:
To me NetBSD and Haiku would be awesome!